src/dbos-executor.ts

```diff
@@ -617,54 +720,57 @@ export class DBOSExecutor {
     }
     const parentCtx = this.#getRecoveryContext(workflowUUID, wfStatus);
 
-    const wfInfo: WorkflowInfo | undefined = this.workflowInfoMap.get(wfStatus.workflowName);
+    const {wfInfo, config} = this.getWorkflowInfoByStatus(wfStatus);
 
     if (wfInfo) {
       // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      return this.workflow(wfInfo.workflow, { workflowUUID: workflowUUID, parentCtx: parentCtx ?? undefined }, ...inputs);
+      return this.workflow(wfInfo.workflow, { workflowUUID: workflowUUID, parentCtx: parentCtx, classConfig: config }, ...inputs);
     }
 
     // Should be temporary workflows. Parse the name of the workflow.
     const wfName = wfStatus.workflowName;
     const nameArr = wfName.split("-");
     if (!nameArr[0].startsWith(DBOSExecutor.tempWorkflowName)) {
+      // CB - Doesn't this happen if the user changed the function name in their code?
```

Comment:
Yeah if this happens, it means there's a pending workflow that cannot be recovered so we have to throw an error here. This should never happen on apps deployed to DBOS Cloud because we always run the correct code version.